File: lessons/compiled_lang/cs/src/d10.input.cs
using System;

class D10Input
{
    static void Main()
    {
        Console.Write("Input a character: ");
        // Console.In.Read() reads exactly one character's code from
        //  stdin as an int - unlike Console.ReadKey(), which requires a
        //  real interactive console and fails/hangs on redirected stdin.
        char character = (char)Console.In.Read();
        Console.WriteLine($"You entered: >>|{character}|<<.");
    }
}

File: lessons/compiled_lang/rust/src/d10.input.rs
use std::io::{self, Read, Write};

fn main() {
    print!("Input a character: ");
    io::stdout().flush().unwrap();

    // Exactly one byte, not a whole line - the piped input carries a
    //  trailing newline (see d00.input.rs's read_line), which a
    //  single-character read must leave unconsumed on the stream rather
    //  than accidentally capturing.
    let mut buf = [0u8; 1];
    io::stdin().read_exact(&mut buf).unwrap();
    let character = buf[0] as char;

    println!("You entered: >>|{character}|<<.");
}
